jupyter_server/services/nbconvert/handlers.py

```diff
@@ -28,22 +33,22 @@ async def get(self):
         # Some exporters use the filesystem when instantiating, delegate that
         # to a thread so we don't block the event loop for it.
         exporters = await run_sync(base.get_export_names)
-        for exporter_name in exporters:
-            try:
-                async with LOCK:
+        async with self._exporter_lock:
```

Comment:
Moved acquisition of the lock outside the loop so we aren't acquiring and re-acquiring the same lock for each exporter.